### DISCUSSION TITLE
Index prices with multiple currencies

### DISCUSSION THREAD

**Discussion by rauldeheer** [meilisearch/product#463] [Feedback & Feature Proposal]
Hi!

My me and my team are working on a website that should support all currencies. We basically need to be able to show our website in all currencies, including the search page and results. How should we approach this with Meilisearch? We currently have a single `products` index with all product information.

We were unable to find a proper solution on the internet. Is the only solution to create one index for each currency? For example:

1. `products_usd`
2. `products_eur`
3. `products_aud`

Etc.

Thanks for helping in advance.

**Comment by gmourier** [comment]
Hi @rauldeheer 👋

Why not store the currencies inside a product document?

```json
    {
         "id": 1,
         "name": "Product name",
         "usd_price" : 1,
         "eur_price": 2,
         "aud_price": 3
    }
```

**Reply by rauldeheer** [reply]
Hey. Thanks for your answer. This is possible, but not really convenient. The way I suggested is not convenient either. I just cannot believe that there wouldn't be a special way to treat this, as most webshops will have this problem I believe?

**Reply by gmourier** [reply]
Can you please elaborate on why it's not convenient to have the currencies by document @rauldeheer? Thanks!

Up to now, it hasn't been a problem, I'd love to know more about it.

**Reply by rauldeheer** [reply]
Well it may be a 'luxury problem', but how I would like to see it is a bit different. I can image that it would sometimes be convenient to be able to have some sort of hook functions which can be enabled on specific document fields. For example lets take the following document from index `products`:

```json
{
    "id": 1,
    "name": "Product SALE!",
    "price" : 1
}
```

Let's say that I would like to filter certain words from the `name` field. I imagine you can make some sort of PrePostHook that removes the word SALE from all product names. Now lets take my example with the same document:

I would like the `price` field to be the price of the current request (any currency). So I imagine you would then make some sort of hook BEFORE the document(s) are retrieved. So I would then look at the pricing, perform some sort of small calculation on the pricing to convert it to the right currency and finally return the results. 

If I took your example I would have to do the following:

1. Store ALL currencies in the document (100+ currencies in our case).
2. Update the WHOLE index every x hours to update all prices according to our base currency (EUR).

**Comment by curquiza** [comment]
I've just transferred this discussion to the right repo so that the product team can take care of it the right way 😇

**Reply by rauldeheer** [reply]
Thanks! Do you have a link for me perhaps?

**Reply by curquiza** [reply]
It has already been put in the right repo: it was in the `meilisearch` repo before, and now it's in the `product` repo. GitHub handles the redirection. No link to share and nothing to do on your side 😊 
My message was just a "For Your Information" message, in case you wonder why your discussion has been transfered.
